src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -52,18 +52,31 @@ public Connection(Uri url, ILoggerFactory loggerFactory)
 
         public Task StartAsync(ITransport transport, HttpClient httpClient)
         {
-            _startTask = StartAsyncInternal(transport, httpClient);
-            return _startTask;
-        }
-
-        private async Task StartAsyncInternal(ITransport transport, HttpClient httpClient)
-        {
             if (Interlocked.CompareExchange(ref _connectionState, ConnectionState.Connecting, ConnectionState.Initial)
                 != ConnectionState.Initial)
             {
-                throw new InvalidOperationException("Cannot start a connection that is not in the Initial state.");
+                return Task.FromException(
+                    new InvalidOperationException("Cannot start a connection that is not in the Initial state."));
             }
 
+            StartAsyncInternal(transport, httpClient)
+                .ContinueWith(t =>
+                {
+                    if (t.IsFaulted)
```

Comment:
I think it's better to be safe than sorry and go for full completeness. Especially since if `StartAsyncInternal` ever throws an `OperationCancelledException` it will result in it being marked cancelled, not faulted.

This would actually make a useful helper someday (`.ContinueBySignallingTcs` or something). But we're not at the point of needing it in a helper (and we may never be ;)). But just in case we ever did, or we looked at this code and reused it somewhere, I'd rather be complete ;)